quinn-proto/src/connection/mod.rs

```diff
@@ -2401,12 +2413,12 @@ where
                     if self.side.is_server() && self.peer_params.stateless_reset_token.is_none() {
                         // We're a server using the initial remote CID for the client, so let's
                         // switch immediately to enable clientside stateless resets.
-                        debug_assert_eq!(self.rem_cid_seq, 0);
+                        debug_assert_eq!(self.rem_cids.rem_cid_seq(), 0);
                         self.update_rem_cid().unwrap();
-                    } else if frame.retire_prior_to > self.rem_cid_seq {
-                        // If our current CID is earlier than the first unretired one we must not
-                        // have adopted the one we just got, so it must be stored in rem_cids, so
-                        // this unwrap is guaranteed to succeed.
+                    } else if update_reset_token && !self.rem_cids.validate_rem_cid() {
```

Comment:
Would this be sufficient? I can't see how we'd end up with a retired active remote CID otherwise.
```suggestion
                    } else if !self.rem_cids.validate_rem_cid() {
```

Reply:
Right. Checking `update_reset_token ` seems not necessary.